test/unit/settings_test.rb

```diff
@@ -169,6 +179,26 @@ def test_require_cc_on_signup_visible_ui_switch_on_rolling_updates
     assert settings.monthly_billing_enabled
   end
 
+  test 'empty values sanitized for non-null columns' do
+    settings.update(public_search: true)
+    assert settings.reload.public_search
+
+    settings.update(public_search: "")
+    assert settings.reload.public_search
```

Comment:
The update was skipped? Is this the intention? Maybe rename the test to something like "empty values are skipped for non-null columns". I wonder how will that work during creation of `settings` though.

Reply:
Good question :sweat_smile:

Reply:
In https://github.com/3scale/porta/pull/3818/commits/286bf51381a4fe578b96c619761aafff301dca4a I moved this filtering to `assign_attributes`, now it should work for both `.create`, `#update` and other public methods.

Reply:
That was just a question. I don't know if it makes sense to sanitize for create. Because if we filter out a required non-null attribute, the query is supposed to fail anyway. But I'm fine if you consider this a better approach.